app/src/test/java/com/example/universitieslist/data/remote/repository/UniversityRepositoryImplTest.kt

```diff
@@ -0,0 +1,80 @@
+package com.example.universitieslist.data.remote.repository
+
+import com.example.universitieslist.data.model.UniversityResponse
+import com.example.universitieslist.data.remote.ApiService
+import com.example.universitieslist.util.Constants.COUNTRY_NAME
+import com.example.universitieslist.util.Constants.UNABLE_TO_FETCHING
+import com.example.universitieslist.util.Constants.UNKNOWN_ERROR
+import com.example.universitieslist.util.FetchUniversitiesException
+import com.example.universitieslist.util.mockUniversityResponses
+import io.mockk.coEvery
+import io.mockk.mockk
+import kotlinx.coroutines.flow.toList
+import kotlinx.coroutines.test.runTest
+import okhttp3.ResponseBody
+import org.junit.Assert.assertEquals
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Test
+import retrofit2.Response
+import kotlin.test.assertFailsWith
+
+class UniversityRepositoryImplTest {
+
+    private val mockApiService = mockk<ApiService>()
+    private lateinit var universityRepositoryImpl: UniversityRepositoryImpl
+
+    @Before
+    fun setUp() {
+        universityRepositoryImpl = UniversityRepositoryImpl(mockApiService)
+    }
+
+    @Test
+    fun `getUniversities list when response is successful`() = runTest {
+        val countryName = COUNTRY_NAME
+        val expectedUniversityList = mockUniversityResponses
+        val fakeResponse = Response.success(expectedUniversityList)
+
+        coEvery { mockApiService.getUniversities(eq(countryName)) } returns fakeResponse
+
+        val resultList = universityRepositoryImpl.getUniversities(countryName).toList()
+
+        assertTrue("Result list should not be empty", resultList.isNotEmpty())
+        assertEquals(
+            "The returned university list should match the expected mock data",
```

Comment:
Is there any reason for this message in the `assert` functions?

Reply:
yeah, kind of comments about what they do, I can remove it if it's not okay with you.

Reply:
I believe if your test function is descriptive, then, that should suffice.

Reply:
yeah, let me remove them